src/org/mozilla/javascript/optimizer/Block.java

```diff
@@ -530,6 +530,7 @@ private static int findExpressionType(OptFunctionNode fn, Node n,
             case Token.ARRAYCOMP:
             case Token.ARRAYLIT:
             case Token.OBJECTLIT:
+            case Token.BIGINT:
```

Comment:
It seems that there is still a problem with optimization.
I will fix it and redo the measurement.
```js
// $ java -jar buildGradle/libs/rhino-1.7.14-SNAPSHOT.jar -version 200 -opt 9

const one = 1n;
(() => 1n - one)(); // js: uncaught JavaScript runtime exception: TypeError: Cannot convert BigInt to a number
```